std/rb.zig

```diff
@@ -0,0 +1,523 @@
+const Color = enum(u1) {
+    Black,
+    Red,
+};
+const Red = Color.Red;
+const Black = Color.Black;
+
+pub const Node = struct {
+    left: ?*Node,
+    right: ?*Node,
+    parent_and_color: usize,
```

Comment:
If we're going to stuff extra bits into a pointer, we should assert the alignment is big enough to fit, and the type should be the pointer type, to guarantee the alignment.

```zig
parent_and_color: *Node,

comptime {
    assert(@alignOf(*Node) >= 2);
}
```

(You'll have to change some of the `@intToPtr` / `@ptrToInt` code elsewhere to match.)

Reply:
I changed this to *Node, but i'm thinking i want it back as usize (but still with the assert), cause it isn't really a pointer when I am stuffing stuff into the least significant bit.

Reply:
I thought about it more, and I think you're right. Let's just have a `comptime assert(@alignOf(*Node) >= 2);` above the line where you do the `@intToPtr`.